apps/jobs/views.py

```diff
@@ -624,6 +624,149 @@ def leaderboard(request, challenge_phase_split_id):
     return paginator.get_paginated_response(response_data)
 
 
+@api_view(["GET"])
+@throttle_classes([AnonRateThrottle])
+def complete_leaderboard(request, challenge_phase_split_id):
+    """
+    Returns complete leaderboard for a corresponding Challenge Phase Split
+
+    - Arguments:
+        ``challenge_phase_split_id``: Primary key for the challenge phase split for which leaderboard is to be fetched
+
+    - Returns:
+        All Leaderboard entry objects in a list
+    """
+    # check if the challenge exists or not
+    try:
+        challenge_phase_split = ChallengePhaseSplit.objects.get(
+            pk=challenge_phase_split_id
+        )
+    except ChallengePhaseSplit.DoesNotExist:
+        response_data = {"error": "Challenge Phase Split does not exist"}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    # Get the leaderboard associated with the Challenge Phase Split
```

Comment:
Please add these conditions so that only challenge host can access this API -
```
    challenge_obj = challenge_phase_split.challenge_phase.challenge

    if not is_user_a_host_of_challenge(request.user, challenge_obj.pk):
        response_data = {"error": "Sorry, you are not authorized to make this request!"}
        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
```